app/src/main/java/com/example/sharewhatyoucanproject/HomeFragment.kt

```diff
@@ -0,0 +1,133 @@
+package com.example.sharewhatyoucanproject
+
+import android.Manifest
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.os.Bundle
+import android.os.Looper
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.TextView
+import android.widget.Toast
+import androidx.core.app.ActivityCompat
+import androidx.fragment.app.Fragment
+import com.example.sharewhatyoucanproject.DonorActivity
+import com.google.android.gms.location.LocationCallback
+import com.google.android.gms.location.LocationResult
+import com.google.android.gms.location.LocationServices
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.firestore.GeoPoint
+import com.google.type.LatLng
+
+class HomeFragment : Fragment() {
+    var uploadlayout: Button? = null
+    var nameofuser: TextView? = null
+    lateinit var checklocation: Button
+    var latitude: Double = 44.5581222
+
+    // -123.2734361
+    var longitute: Double = -123.2734361
```

Comment:
Why the extra comment?  Doesn't seem to add anything.

Also why this magic number?